persistent/Database/Persist/SqlBackend/Internal/MkSqlBackend.hs

```diff
@@ -28,7 +27,7 @@ data MkSqlBackendArgs = MkSqlBackendArgs
     , connInsertSql :: EntityDef -> [PersistValue] -> InsertSqlResult
     -- ^ This function generates the SQL and values necessary for
     -- performing an insert against the database.
-    , connStmtMap :: IORef (Map Text Statement)
+    , connStmtMap :: StatementCache
```

Comment:
Hmm. We may be able to avoid making this a breaking change, if there's a way to go from `IORef (Map Text Statement)` to a `StatementCache` in the function that translates a `MkSqlBackendArgs` into a `SqlBackend`.

Reply:
OK, yeah, this can definitely be a minor version bump! Which means there's no urgency to figure out the other breaking changes and try to bundle them together. I think this is the only breaking change in the PR.

Reply:
I want to mention that I'd like to ensure that _alternative_ statement caches can be provided too. In particular, the end goal here is that we could for example provide counters of the statement cache to detect memory leaks (real situation that can occur for long-lived postgresql connections currently AFAICT). Or as another example, we might want to provide an LRU version of the statement cache that can evict. That's why this was introduced as a breaking change. That could technically be deferred til a later date.

Reply:
You should be able to write `setConnStatementCache :: StatementCache -> SqlBackend -> SqlBackend`, without breaking this.

In hindsight, I wish I made this a `Maybe` field - or even omit it from the list entirely. There's an obvious default, though it does mean that `mkSqlBackend :: MkSqlBackendArgs -> IO SqlBackend` needs to be done to produce the cache instead.

So, for a 2.13 release, can we avoid breaking this?

Then, for 2.14, we can either remove this (and only use the `setConnStatementCache`) or make it a `Maybe` field, and *also* make `mkSqlBackend` in `IO`.